src/app/shared/products-filter/products-filter.pipe.ts

```diff
@@ -0,0 +1,16 @@
+import {Pipe, PipeTransform} from '@angular/core';
+import {Product} from '../products/product.interface';
+
+@Pipe({
+    name: 'productsFilter',
+    standalone: true,
+})
+export class ProductsFilterPipe implements PipeTransform {
+    transform(products: Product[] | null, searchName: string): Product[] | null {
+        if (products === null) {
+            return null;
+        }
+
+        return products?.filter(product => product.name.includes(searchName));
```

Comment:
А `?.` тут нужен? Выше вроде проверили на причастность к `null` 🤔 

А вообще я порекомендовал бы так сделать:
```suggestion
        return products && products.filter(product => product.name.includes(searchName));
```